spec/highlighting.spec.js

```diff
@@ -54,4 +76,237 @@ describe('Highlighting', function () {
     })
   })
 
+  describe('highlightSupport', () => {
```

Comment:
All these tests never test if we actually to the right thing! You only check that the index positions are the same, we do not know that the right characters are selected.

Reply:
Tests now check for the text equivalent of the ranges. See also comment below.